blockchain/genesis.go

```diff
@@ -307,10 +308,21 @@ func (g *Genesis) ToBlock(baseStateRoot common.Hash, db database.DBManager) *typ
 		db = database.NewMemoryDBManager()
 	}
 	stateDB, _ := state.New(baseStateRoot, state.NewDatabase(db), nil, nil)
+	rules := params.Rules{}
+	if g.Config != nil {
+		rules = g.Config.Rules(new(big.Int).SetUint64(g.Number))
+	}
 	for addr, account := range g.Alloc {
-		if len(account.Code) != 0 {
-			originalCode := stateDB.GetCode(addr)
+		originalCode := stateDB.GetCode(addr)
+		if _, ok := types.ParseDelegation(account.Code); ok && rules.IsPrague {
+			stateDB.SetCodeToEOA(addr, account.Code, rules)
+		} else if len(account.Code) == 0 && len(account.Storage) != 0 && rules.IsPrague {
+			stateDB.CreateEOA(addr, false, accountkey.NewAccountKeyLegacy())
+		} else if len(account.Code) != 0 {
+			stateDB.CreateSmartContractAccount(addr, params.CodeFormatEVM, rules)
 			stateDB.SetCode(addr, account.Code)
+		}
+		if len(account.Code) != 0 {
```

Comment:
Can't you sum it up in line 321?
I mean
- if len(account.Code) == 0 && len(account.Storage) != 0 && rules.IsPrague
- else if len(account.Code) != 0
  - if _, ok := types.ParseDelegation(account.Code); ok && rules.IsPrague {SetCodeToEOA}
  - else {create && setcode}
  - if originalCode != nil

Reply:
That's possible but I don't know if it's more readable.

Reply:
It was a very detailed performance point of view. As you said, it doesn't improve readability. How about changing it to a switch statement with explanatory variables?

Reply:
I'll try it.